core/test/presentation/views/text/rich_text_builder_test.dart

```diff
@@ -0,0 +1,208 @@
+  );
+  Widget buildRichText({
+    required String textOrigin,
+    required String wordToStyle,
+    String? preMarkedText,
+    bool ensureHighlightVisible = false,
+  }) => MaterialApp(home: Scaffold(
+    body: RichTextBuilder(
+      textOrigin: textOrigin,
+      wordToStyle: wordToStyle,
+      styleOrigin: normalTextStyle,
+      styleWord: highlightTextStyle,
+      preMarkedText: preMarkedText,
+      ensureHighlightVisible: ensureHighlightVisible,
+    ),
+  ));
+
+  Finder findNormalStyledText(String text, {int occurence = 1}) {
+    final normalSpan = TextSpan(text: text, style: normalTextStyle);
+    return find.byWidgetPredicate((widget) {
+      return widget is RichText
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.contains(normalSpan)
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.countOccurences(normalSpan) == occurence;
+    });
+  }
+
+  Finder findHighlightStyledText(String text, {int occurence = 1}) {
+    final highlightSpan = TextSpan(text: text, style: highlightTextStyle);
+    return find.byWidgetPredicate((widget) {
+      return widget is RichText
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.contains(highlightSpan)
+        && ((widget.text as TextSpan).children!.first as TextSpan).children!.countOccurences(highlightSpan) == occurence;
+    });
+  }
+
+  group('rich text builder test:', () {
+    testWidgets(
+      'should not highlight any text '
+      'when wordToStyle is empty '
+      'and preMarkedText is null',
+    (tester) async {
+      // arrange
+      const textOrigin = 'This is a test';
+      final richText = buildRichText(
+        textOrigin: textOrigin,
+        wordToStyle: '',
+      );
+      
+      // act
+      await tester.pumpWidget(richText);
+      
+      // assert
+      expect(
+        findNormalStyledText(textOrigin),
+        findsOneWidget,
+      );
+    });
+
+    testWidgets(
+      'should highlight text with wordToStyle '
+      'when wordToStyle is not empty '
+      'and preMarkedText is null',
+    (tester) async {
+      // arrange
+      const textOrigin = 'This is a test of test';
+      const wordToStyle = 'test';
+      final richText = buildRichText(
+        textOrigin: textOrigin,
+        wordToStyle: wordToStyle,
+      );
+      final numberOfHighlights = wordToStyle.allMatches(textOrigin).length;
+      final normalStyledTexts = textOrigin
+        .split(wordToStyle)
+        ..removeWhere((text) => text.isEmpty);
+      
+      // act
+      await tester.pumpWidget(richText);
+      
+      // assert
+      for (final text in normalStyledTexts) {
+        expect(
+          findNormalStyledText(
+            text,
+            occurence: normalStyledTexts.countOccurences(text),
+          ),
+          findsOneWidget,
+        );
+      }
+      expect(
+        findHighlightStyledText(wordToStyle, occurence: numberOfHighlights),
+        findsOneWidget,
```

Comment:
why only one, I think 2

Reply:
The occurrences is 2, but the whole text is only 1.